rs/ledger_suite/icrc1/ledger/src/lib.rs

```diff
@@ -704,25 +707,40 @@ impl<Tokens: TokensType> Ledger<Tokens> {
     /// The canister code must call set_certified_data with the value this function returns after
     /// each successful modification of the ledger.
     pub fn root_hash(&self) -> [u8; 32] {
-        self.construct_hash_tree().digest().0
+        self.construct_hash_tree().digest()
     }
 
-    pub fn construct_hash_tree(&self) -> MixedHashTree {
+    pub fn construct_hash_tree(&self) -> HashTree {
         match self.blockchain().last_hash {
-            Some(hash) => {
+            Some(last_block_hash) => {
                 let last_block_index = self.blockchain().chain_length().checked_sub(1).unwrap();
-                MixedHashTree::Fork(Box::new((
-                    MixedHashTree::Labeled(
-                        Label::from("last_block_index"),
-                        Box::new(MixedHashTree::Leaf(last_block_index.to_be_bytes().to_vec())),
-                    ),
-                    MixedHashTree::Labeled(
-                        Label::from("tip_hash"),
-                        Box::new(MixedHashTree::Leaf(hash.as_slice().to_vec())),
-                    ),
-                )))
+                let last_block_index_label = Label::from("last_block_index");
+
+                #[cfg(feature = "icrc3-compatible-data-certificate")]
+                {
+                    let last_block_hash_label = Label::from("last_block_hash");
+                    let mut last_block_index_encoded = Vec::new();
```

Comment:
nit: `last_block_index` here is a `u64`, and the largest amount of bytes a `leb128`-encoded `u64` can occupy is 10, so a minor optimization here would be to use `Vec::with_capacity(10)` to [potentially avoid some reallocations](https://doc.rust-lang.org/std/vec/struct.Vec.html#capacity-and-reallocation) (for an example see [here](https://sourcegraph.com/github.com/dfinity/ic@48dd3fdce379eab7e66ea30852a02e6a59b43fbc/-/blob/rs/registry/canister/src/certification.rs?L27-32)).